code/modules/antagonists/antagonist_helpers.dm

```diff
@@ -17,7 +17,19 @@ proc/get_all_gangs()
 	for (var/datum/antagonist/gang_leader/antag_datum in get_all_antagonists(ROLE_GANG_LEADER))
 		. += antag_datum.gang
 
-	return .
+/** Returns the mind of this mind's master.
+ * If a specific antagonist role ID is provided, then the master of the antagonist datum of that ID belonging to this mind will be returned.
+ * If no antagonist role ID is provided, then the master of the newest antagonist datum will be returned.
+ */
+/datum/mind/proc/get_master(antagonist_role_id)
+	if (antagonist_role_id)
+		var/datum/antagonist/subordinate/antagonist_role = src.get_antagonist(antagonist_role_id)
+		if (antagonist_role?.master)
+			. = antagonist_role.master
+		return
+
+	for (var/datum/antagonist/subordinate/antagonist_role in src.antagonists)
+		. = antagonist_role.master // Return the master of the newest antagonist datum in `antagonists`.
```

Comment:
is the newest at the end or something?

Reply:
That is how lists work, to my knowledge. The most recently instantiated antagonist datum _should_ append itself to the end of the `antagonists` list on the owner mind.